### PR TITLE
fix(windows): enable Websh for any enabled local user

### LINKED ISSUE #302: fix(windows): grant Administrator (RID 500) a login shell regardless of Get-LocalUser Enabled flag
## Problem

End-to-end testing on Windows hosts revealed a registration-environment difference:

- **AWS EC2 Windows instances:** websh access to `Administrator` works correctly.
- **Default Windows 10 / 11 laptops:** websh refuses the session because the synced `SystemUser` for Administrator shows `login_enabled=false`. Operators see `SYSTEM_USER_ACCESS_DENIED`.

### Environment difference (Windows default behavior)

| Environment | Built-in Administrator default | `Get-LocalUser Administrator \| Select Enabled` |
|---|---|---|
| AWS EC2 Windows AMI | **Enabled** (activated with password at instance launch) | `True` |
| Default Windows 10 / 11 laptop | **Disabled** (must be explicitly activated with `net user Administrator /active:yes`) | `False` |

This is the documented Windows security default — not an environment misconfiguration.

### Root cause inside alpamon

`pkg/runner/commit_windows.go:50-62` parses `Get-LocalUser | Select Name,SID,Enabled | ConvertTo-Csv` and decides whether to grant Administrator a login shell:

```go
// line 50
enabled := strings.EqualFold(fields[2], "True")
// lines 57-62
shell := ""
if enabled && uid == 500 {
    shell = utils.DefaultShell()  // powershell.exe
}
```

- AWS EC2: `enabled=true, uid=500` → `shell="powershell.exe"`
- Laptop: `enabled=false, uid=500` → `shell=""` (empty)

The sync payload (`pkg/runner/commit_types.go:99-109` `UserData.Shell`) is therefore an empty string on locked-Administrator laptops. By the agent ↔ backend contract, an empty `shell` is treated as "this user cannot log in", so the resulting `SystemUser.login_enabled` is `false` and websh refuses the session. **The empty-shell signal is what blocks websh — and that signal originates here in alpamon.**

The line 25 comment in `commit_windows.go` acknowledges the current limitation: "until credential-based demotion is implemented, only Administrator (RID 500) gets shell; other Enabled users can be granted login shells [in the future]".

### Why the `Enabled` check is meaningless today

`pkg/utils/privilege_windows.go` is currently a stub — **every command runs under the SYSTEM service account regardless of which user "owns" the websh session**. Whether Administrator is enabled or disabled at the OS level has zero effect on the executed permissions. The `enabled && uid == 500` guard appears to restrict login, but it is purely cosmetic: it neither escalates nor restricts what the shell can actually do. The check would only become meaningful if and when privilege demotion (e.g., `CreateProcessAsUser` with a logon token) is implemented.

### Note on "Users system user" sometimes mentioned in reports

`BUILTIN\Users` is a **Windows group**, not a user. alpamon's `getGroupData()` collects it separately from `getUserData()`. Groups do not carry a `login_enabled` field and are unrelated to this websh failure. A unified UI list of users + groups can cause operator confusion; track that as a separate UI ticket if it persists.

## One-line summary

> alpamon trusts the PowerShell `Enabled` property to decide whether Administrator gets a login shell. Default-locked Administrator on Windows laptops produces `shell=""` in the sync payload, which the backend correctly treats as `login_enabled=false`, which websh correctly refuses. Removing the `Enabled` predicate fixes the regression without weakening security, because all commands already run as SYSTEM.

## Expected behavior

- Websh access as `Administrator` succeeds on Windows hosts regardless of whether the built-in Administrator account is enabled at the OS level.
- AWS EC2 Windows behavior is unchanged (no regression).
- Operators are informed (via `docs/windows.md`) that the displayed `Administrator` session actually runs under `NT AUTHORITY\SYSTEM` until privilege demotion is implemented.

## Acceptance criteria

1. `pkg/runner/commit_windows.go:60` — drop the `enabled &&` predicate so Administrator (RID 500) always receives a login shell:

   ```go
   // before
   if enabled && uid == 500 {
       shell = utils.DefaultShell()
   }
   // after
   if uid == 500 {
       shell = utils.DefaultShell()
   }
   ```

2. Unit test in `pkg/runner/commit_windows_test.go` (create the file if absent). Three cases:
   - Case A: `enabled=true, uid=500` → `shell="powershell.exe"` (existing behavior preserved).
   - Case B: `enabled=false, uid=500` → `shell="powershell.exe"` (the new behavior — regression guard for laptops).
   - Case C: `enabled=true, uid=1001` (a regular non-Administrator local user) → `shell=""` (unchanged; non-Admin users still receive no login shell while privilege demotion is unavailable).

3. `docs/windows.md` — add a "Permissions & identity" subsection:

   > **Permissions & identity.** Until alpamon implements credential-based privilege demotion, all websh commands execute under `NT AUTHORITY\SYSTEM` regardless of the displayed session user. This means:
   >
   > - Administrator websh works even when the built-in Administrator account is disabled at the Windows OS level.
   > - Commands run with full SYSTEM privileges. Configure your Alpacon role and policy settings accordingly.
   > - Running `whoami` inside the session may print `nt authority\system` rather than `administrator` — this is expected.

4. `go test ./... -p 1` passes on the `windows-latest` CI matrix.

## Out of scope

- **Privilege demotion implementation** (`CreateProcessAsUser` with a logon token). This is the architectural follow-up that would make the `Enabled` check meaningful again. Separate epic — not blocking this bug fix.
- **A UI banner stating "this session actually runs as SYSTEM"** — UX follow-up.
- **Granting non-Administrator enabled users a login shell** — explicitly deferred until privilege demotion lands (line 25 comment in `commit_windows.go` reflects this).
- **Hiding `BUILTIN\Users` and other groups from the system user list in the UI** — separate UI ticket if it confuses operators.

## Operator workaround (until the fix ships)

On the affected laptop, run from an elevated PowerShell:

```powershell
net user Administrator /active:yes
net user Administrator <strong_password>
```

Wait for one alpamon sync cycle (~5 minutes) or restart the alpamon service. Verify in the Alpacon UI that Administrator now shows `login_enabled=true`, then re-try websh.

## Verification (manual)

1. Apply the fix on a fresh `windows-latest` runner and confirm the new unit test cases pass in CI.
2. Install the patched alpamon on a Windows 10 / 11 laptop where Administrator is disabled (Windows default). Register and wait for the first sync.
3. Alpacon admin UI → server's system users list → confirm Administrator shows `login_enabled=true`.
4. Open websh as Administrator → expect a PowerShell prompt with no `SYSTEM_USER_ACCESS_DENIED` error.
5. Run `whoami` inside the session — expect `nt authority\system` (documents the current architecture for operators).
6. Re-test on an AWS EC2 Windows instance to confirm no regression (Administrator still works there).

## References (alpamon-side)

- `pkg/runner/commit_windows.go:21-78` — Windows user collection via `Get-LocalUser`
- `pkg/runner/commit_types.go:99-109` — `UserData` sync payload struct (`Shell` field is the contract surface)
- `pkg/utils/privilege_windows.go` — privilege demotion stub (the reason the `Enabled` check is currently moot)
- Recent commit `71ab1e0` — "replace deprecated wmic with PowerShell Get-LocalUser/Get-LocalGroup" (introduced the parsing path that this issue patches)

### PR DESCRIPTION
**Pull request by geunwoonoh**
## Summary

- Restore Websh access as **Administrator** on default Windows 10 / 11 laptops where the built-in account is OS-level disabled (closes #302).
- Extend the same shell-granting behavior to **every enabled local user**, so laptop operators can use Websh under their own account rather than activating Administrator out-of-band.
- Tighten the local-user CSV parser (encoding/csv, RID-0 skip) so the widened predicate cannot silently emit malformed records.
- Document the resulting SYSTEM-execution trade-off so operators understand the runtime identity model.

## Background

End-to-end testing surfaced an environment-shaped Websh regression on Windows:

| Environment | Built-in Administrator | `Get-LocalUser Administrator \| Select Enabled` | Websh as Administrator |
|---|---|---|---|
| AWS EC2 Windows AMI | Enabled at launch | `True` | works |
| Windows 10 / 11 laptop | **Disabled** (Microsoft default) | `False` | **refused** with `SYSTEM_USER_ACCESS_DENIED` |

The refusal originated in alpamon, not the backend. `pkg/runner/commit_windows.go:60` gated Administrator's login shell on the `Get-LocalUser` `Enabled` column. Default-locked Administrator on laptops therefore synced as `shell=""`, which `alpacon-server`'s `compute_login_enabled` correctly maps to `login_enabled=false`, which Websh correctly refuses.

The `Enabled && uid == 500` predicate was a cosmetic check, not a security boundary: `pkg/utils/privilege_windows.go` is a no-op stub, so every Websh command on Windows already runs as `LocalSystem` regardless of the OS-level Administrator state. Removing the predicate restores parity between EC2 and laptops without weakening the actual security model.

After landing the narrow #302 fix, the same reasoning applies to non-Administrator local users: gating their shell on `Enabled` does not restrict real authority either. The motivating workflow (a laptop operator using Websh under their own account) is only fully unblocked when the predicate covers every enabled local user, not just RID 500. Alpacon RBAC remains the single authorization surface; alpamon does not add a second restriction on top of it.

## What changed

| Area | File | Change |
|---|---|---|
| Predicate | `pkg/runner/commit_windows.go` | `if enabled \|\| uid == 500 { shell = utils.DefaultShell() }`. Extracted into a pure `parseGetLocalUserCSV` so cases are unit-testable. |
| Parser robustness | `pkg/runner/commit_windows.go` | `parseCSVLine` → `encoding/csv` (RFC 4180); skip rows where `ridFromSID` returns 0. |
| Tests | `pkg/runner/commit_windows_test.go` (new) | Windows-only table-driven test with 11 sub-cases covering the predicate's both branches and parser edge cases. |
| Docs | `docs/windows.md` | New `### Permissions and identity` subsection covering SYSTEM execution, Websh-eligibility roster as security posture, and audit attribution. |

## Commits

The PR is intentionally split into reviewable, revertable units.

1. **`8af46bd  fix(windows): grant Administrator RID 500 a login shell regardless of Get-LocalUser Enabled flag`**
   The narrow #302 fix. Extracts `parseGetLocalUserCSV` from `getUserData`; drops the `enabled &&` predicate so RID 500 always receives a login shell. Adds the initial 8-case test file.

2. **`2a68524  docs(windows): refresh getUserData comment and capitalize Websh`**
   Review feedback: removes the stale "When credential-based demotion is implemented, other Enabled users can be granted login shells" sentence (which would have contradicted commit 3 anyway) and capitalizes `Websh` per CLAUDE.md conventions.

3. **`c5a0dd1  feat(windows): grant a login shell to every enabled local user`**
   Widens the predicate to `enabled || uid == 500`. Updates Case C, adds Case D (disabled non-admin regression guard), and generalizes the docs subsection from "Administrator only" to "any enabled local user." This is the policy decision — alpamon stops being a second authorization layer; Alpacon RBAC is the boundary.

4. **`bc1ad5d  fix(windows): tighten local-user CSV parser robustness`**
   Post-review defense-in-depth (W1, W2 from `/alpacax:review`): replace the hand-rolled `parseCSVLine` with `encoding/csv` to handle RFC 4180 escaping; skip rows where `ridFromSID == 0` so a parse failure cannot produce a `UID=0` Websh-enabled record. Symmetric fix in `getGroupData`. Two new test cases pin both behaviors.

5. **`0ead41f  docs(windows): document Websh persistence pivot and audit attribution`**
   Post-review operator-facing notes (I1, I2). Adds a new bullet stating that the *roster* of Websh-eligible users is part of the security posture (a local admin can re-enable a disabled account such as `Guest` / `WDAGUtilityAccount` and turn it into a persistence channel — alarm on unexpected new `login_enabled` users in the Alpacon audit feed). Extends the `whoami` paragraph with attribution guidance (correlate Alpacon audit log with Windows event log).

### Code-level vulnerabilities introduced

None. PowerShell command is a fixed string literal (no user input interpolation); `True`/`False` parsing is locale-invariant (`bool.ToString()` uses invariant culture); sync path is serialized by `syncMutex`.

### Policy trade-off (deliberately accepted)

Before this PR, alpamon's `if uid == 500` predicate served as a second defense layer that capped Websh-as-anyone to Websh-as-Administrator on Windows. After this PR, alpamon emits `shell="powershell.exe"` for every enabled local user; Alpacon RBAC is the sole authorization surface.

This is documented explicitly in `docs/windows.md` "Permissions and identity" (commit 5):

- **Audit-label vs. permission-boundary distinction**: Granting `alice` Websh access does not run commands as `alice` — they run as SYSTEM with `alice` as the audit label. Operators must treat any Websh-enabled local user as effectively SYSTEM.
- **Roster as security posture**: A local admin on the host can re-enable a disabled local account and turn it into a Websh persistence channel. The set of `login_enabled=true` users is part of the security posture, not just an OS state.
- **Attribution**: SYSTEM execution means `whoami` returns `nt authority\system`, not the requested user. Correlating the Alpacon audit log (Alpacon user → target local user) with the Windows event log (SYSTEM-level execution) is required for attribution until privilege demotion ships.

### Server-side dependencies (verified read-only)

A separate read-only audit of `alpacon-server` confirms the load-bearing assumptions hold:

- **Q1 — Session-open re-validation**: `WebshValidationSerializer.validate()` (`websh/api/serializers.py`) calls `server.has_system_user_access()` (`servers/models.py:705-727`), which does a fresh DB lookup of `SystemUser`, checks `login_enabled`, and enforces explicit identity binding (IAM user ID match, Application ID match, or unlinked). No wildcard "any system_user on this server" policy exists. ✅ **PASS**
- **Q2 — `login_enabled` TOCTOU**: `is_access_allowed` (`proc/models.py:192-212`) is evaluated at session-open from a fresh DB read. The session-open → WebSocket-connect window is sub-second and lives in `alpacon-proxy` (separate repo, not verified here). The sync-interval TOCTOU (~5 min) is a pre-existing property of polling sync and applies to every platform, not introduced by this PR. ⚠️ **PARTIAL, not blocking**
- **Q3 — Audit log chain**: `_build_activity_data` (`audit/api/mixins.py:148-185`) records the requesting Alpacon user separately from the target system_user (`object_data.username`). Server platform is not denormalized into the audit document but is derivable via the Session→Server FK. ✅ **SUFFICIENT for forensics**

The Q2 sub-second window and the Q3 denormalization are worth tracking as alpacon-server follow-ups but **do not block this alpamon-side change**.

### Known follow-ups (not in scope of this PR)

1. **`pkg/utils/privilege_windows.go` — privilege demotion** (`CreateProcessAsUser` + logon token). When this lands, the SYSTEM-execution audit-label distinction collapses and operators see real per-user authority. Separate epic.
2. **alpacon-server WebSocket-connect re-validation**: add a `login_enabled` re-check at WebSocket accept time to close the session-open → ws-connect sub-second TOCTOU window. alpacon-server / alpacon-proxy ticket.
3. **alpacon-server audit-log platform denormalization**: add `server.platform` to `Session.get_activity_fields()` for SIEM ergonomics. alpacon-server ticket.

### Automated (this PR)

- [x] `GOOS=windows go build ./...` — clean
- [x] `GOOS=windows go vet ./pkg/runner/...` — clean
- [x] `GOOS=windows go test -c -o /dev/null ./pkg/runner/` — Windows test binary compiles
- [x] `go test ./pkg/runner/... -p 1` (native macOS) — existing tests pass; new `*_windows_test.go` skipped by build tag
- [x] CI `build-and-test-windows` (`.github/workflows/build-and-test.yml:137-182`) will run the full `go test -v ./... -p 1` on `windows-latest`, exercising all 11 sub-cases of `TestParseGetLocalUserCSV`

### Manual (pre-merge or post-merge on a fresh host)

- [ ] Windows 10 / 11 laptop with **Administrator OS-disabled** (Microsoft default):
  - Install the patched alpamon, register, wait one sync (~5 min).
  - Alpacon admin UI → server's system users → `Administrator` shows `login_enabled=true`.
  - Open Websh as `Administrator` → PowerShell prompt, no `SYSTEM_USER_ACCESS_DENIED`.
- [ ] Same laptop, with a personal account `John` enabled (typical operator scenario):
  - Alpacon UI → `John` shows `login_enabled=true`.
  - Open Websh as `John` → PowerShell prompt.
- [ ] Inside both sessions: `whoami` returns `nt authority\system` (documents the SYSTEM execution invariant).
- [ ] **Regression**: AWS EC2 Windows AMI re-test → Administrator Websh still works exactly as before.
- [ ] **Negative**: confirm disabled service accounts (`Guest`, `DefaultAccount`, `WDAGUtilityAccount`) appear with `login_enabled=false` in the Alpacon UI and refuse Websh.

## Out of scope

- **Privilege demotion implementation** (`CreateProcessAsUser` with a logon token). Separate epic. When this lands, the SYSTEM-execution trade-off collapses and the `enabled || uid == 500` predicate would be re-examined under the new model.
- **UI banner stating "this session runs as SYSTEM"** — UX follow-up on the Alpacon console side, not alpamon.
- **Hiding `BUILTIN\Users` and other groups from the system-user list** in alpacon-web. `BUILTIN\Users` is a group, not a user; alpamon already collects it separately. UI consolidation ticket if needed.
- **alpacon-server `compute_login_enabled` changes** — the empty-shell → `login_enabled=false` rule is correct as a general contract; only the alpamon side needed adjustment.
- **CI `windows-latest` matrix work** — already exists (`build-and-test.yml:137-182`); this PR just adds tests that exercise it.

## References

- Issue: #302

### CODE REVIEW

File: pkg/runner/commit_windows.go [hunk 1]
```diff
@@ -18,29 +19,46 @@ func loadShadowData() map[string]shadowEntry {
...
+// execution.
+func parseGetLocalUserCSV(csvData string) []UserData {
 	validShells := loadValidShells()
 	var users []UserData
 
-	for _, line := range strings.Split(string(out), "\n") {
+	for _, line := range strings.Split(csvData, "\n") {
 		line = strings.TrimSpace(line)
 		if line == "" || strings.HasPrefix(line, "\"Name\"") {
 			continue
```
[the remark this answers is not in the document]
Comment (geunwoonoh) [reply pkg/runner/commit_windows.go:60]: Fixed. Moved the header check to after `parseCSVLine` and require an exact match on `Name`/`SID`/`Enabled` columns instead of a raw-line prefix. A local user literally named `Name` is now emitted correctly (covered by a new regression test case in `commit_windows_test.go`).

File: pkg/runner/commit_windows.go [hunk 2]
```diff
@@ -103,6 +126,9 @@ func getGroupData() ([]GroupData, error) {
 		}
 
 		gid := ridFromSID(sid)
+		if gid == 0 {
+			continue
```
[the remark this answers is not in the document]
Comment (geunwoonoh) [reply pkg/runner/commit_windows.go:142]: Fixed in the same commit. Applied the same approach to the two-column group header: parse first, then skip only when `fields[0]=="Name" && fields[1]=="SID"`. A local group literally named `Name` is no longer dropped.

File: pkg/runner/commit_windows_test.go [hunk 3]
```diff
@@ -0,0 +1,187 @@
...
+// local user receives a login shell. Administrator (RID 500) is always
+// granted a shell (issue #302: built-in admin is disabled by default on
+// Windows 10/11 laptops). Every other enabled user is also granted a
+// shell, with all sessions running as LocalSystem until privilege demotion
+// ships. Disabled non-admin users get no shell. The remaining cases guard
+// the parser against header lines, malformed rows, and empty fields.
+func TestParseGetLocalUserCSV(t *testing.T) {
+	const header = "\"Name\",\"SID\",\"Enabled\"\n"
+	const powershell = "powershell.exe"
+	validShells := []string{"powershell.exe", "cmd.exe", "pwsh.exe"}
```
[the remark this answers is not in the document]
Comment (geunwoonoh) [reply pkg/runner/commit_windows_test.go:19]: Hardcoded test values are the right shape here.

The current implementation is appropriate because:
- These tests document the **contract output** that alpacon-server consumes (the synced `UserData.Shell` and `ValidShells` fields), not "whatever `DefaultShell()` happens to return today".
- If a future change updates `utils.DefaultShell()` or `loadValidShells()` to a different value, that change is worth flagging — the test should fail and force a deliberate decision rather than silently absorbing the new value.
- Using `utils.DefaultShell()` / `loadValidShells()` inside the test would make `expected == parseGetLocalUserCSV(...).Shell` essentially tautological, since the parser literally calls `utils.DefaultShell()`. The assertion would no longer verify the actual output bytes.

In short: hardcoded test values catch unintended drift in the production helpers; helper-derived test values would mask it.

**Review by eunyoung14 — APPROVED** [review]
